### PR TITLE
push pylance to pyright 2024.8

### PR DESCRIPTION
**Pull request by heejaechang**
Here's the list of changes to include:

- add `system memory info` in `mem stats`
- Fix early break in loop in `ImportResolver._getModuleNameForImport`
- invalidate imports when they are actually changed.
- rename workspace name if it is changed after workspace is created
- some refactoring on `find configuration file`
- some other misc changes.

### CODE REVIEW

File: packages/pyright-internal/src/analyzer/typeUtils.ts [hunk 1]
```diff
@@ -448,6 +448,15 @@ export interface MapSubtypesOptions {
     retainTypeAlias?: boolean;
 }
 
+export interface MapSubtypesOptions {
```
Comment (debonte) [inline packages/pyright-internal/src/analyzer/typeUtils.ts:451]: This type already exists on line 436 just above here.

Comment (heejaechang) [reply packages/pyright-internal/src/analyzer/typeUtils.ts:451]: oh. I thought I fixed those.

File: packages/pyright-internal/src/analyzer/serviceUtils.ts
Comment (debonte) [inline packages/pyright-internal/src/analyzer/serviceUtils.ts:1]: Looks like this is part of https://github.com/microsoft/pyrx/pull/5431/files which shouldn't be merged into Pyright?

File: packages/pyright-internal/src/analyzer/typeUtils.ts (hunk 1 above)
Comment (heejaechang) [reply packages/pyright-internal/src/analyzer/typeUtils.ts:451]: ah, this is another merge conflict issues.

**Review by rchiodo — APPROVED** [review]

File: packages/pyright-internal/src/analyzer/serviceUtils.ts
Comment (heejaechang) [reply packages/pyright-internal/src/analyzer/serviceUtils.ts:1]: @rchiodo ?

Comment (rchiodo) [reply packages/pyright-internal/src/analyzer/serviceUtils.ts:1]: It looks correct to me. We need it actually or pyright won't build. It was a refactor of moving those functions out of service.ts

**Review by debonte — APPROVED** [review]

**Pull request merged**